CHANGELOG.md

```diff
@@ -8,6 +8,9 @@
 
 ### Changed
 
+- Moved contents of `item`, `catalog`, and `collection` modules into `core` module
+  (top-level package exports remain unchanged) ([#594](https://github.com/stac-utils/pystac/pull/594))
+
```

Comment:
This I think is for sure a breaking change, and if there's a way to keep these imports working (with a deprecation notice) I'd be in favor. I find, for example, that some IDE's auto-import from the deeper path, so folks could be relying on the original import locations without even knowing it.

Reply:
I got things back to using separate `item`, `catalog`, and `collection` modules in the latest push. I was able to resolve most of the import issues by being more careful about using the `TYPE_CHECKING` gate on imports that were only used for type checking, and by using the new module import style.

There was 1 circular dependency that couldn't be cleanly resolved using these strategies, which was between any of `item`, `catalog`, and `collection` and `stac_io`. The root cause is that each of those modules uses `stac_io` to call `stac_io.StacIO.default()` in some methods that have an optional `stac_io` argument, and the `stac_io` module itself depends on `item`, `catalog`, and `collection` as part of [the logic in `stac_object_from_dict`](https://github.com/stac-utils/pystac/blob/e744897c63fd5561040396231f52e1cec398b024/pystac/stac_io.py#L159-L174). I think there's probably a more elegant solution to be had here, but to work around this, I simply added a `default_stac_io` method to `pystac/__init__.py` that those other modules can use to break the circular dependency.